### PR TITLE
Extend Git section

### PR DESCRIPTION
**Pull request by andrejVuk7**
Extend Git section with details from Git workshop. More details can be found here: https://app.productive.io/1-infinum/projects/109718/tasks/task/4489230?filter=NDY0NDcz.

### CODE REVIEW

File: Git/Pull Requests.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Pull requests
+
+Definition and basic rules of a good PR can be found [here](https://infinum.com/handbook/dotnet/team-practices/new-team-members#prs).
+	
+There could also be additional pre-defined rules that every PR needs to satisfy before getting an approval to be merged (all required reviewers have approved the PR, all comments have been resolved...). Also, all potential merge conflicts (concurrent changes of the same resources) should be resolved prior to the merge.
```
Comment (mahrncic) [inline Git/Pull Requests.md:5]: Let's use `etc.` instead of `...`

**Review by mahrncic — CHANGES_REQUESTED** [review]
Great work @andrejVuk7, a lot of useful information is shared in the new pages 😄
I left a couple of comments regarding formatting and wording, so check it out.

**Review by icottiero — CHANGES_REQUESTED** [review]
Like I already told you, kudos for the effort! I have some styling suggestions, and I think there are a couple of grammar oddities, so I suggest running this through Grammarly.

Usually, when I write something for the Handbook, I use VS Code with the Grammarly extension (enabled for MD files, which is for some reason not the default setting). I found that to be much easier to use than c/p to Grammarly and then back :)

**Comment by icottiero** [comment]
@andrejVuk7 almost forgot, please set the target branch to `staging` so that we can check if everything looks good before publishing :)

File: Git/Branching Strategies.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+- feature/add-user-basic-authentication
+- feature/task-344096-add-download-report-endpoint
+- release/v3.1.1
+- hotfix/resolve-user-account-creation-bug
+
+### Examples of branching strategies
+
+#### GitFlow
+
+GitFlow is a branching model for Git which we use here at Infinum. Its main advantage is that it supports projects where deployments are made often. Since we use agile workflow, this goes hand in hand with it. We will cover some of the things about GitFlow in this handbook, but you can find a broader explanation [here](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:21]: Updated the sentence, please check.

File: Git/Branching Strategies.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- ideal for CI/CD (allows all the features to be deployed as soon as they are merged)
+- suitable for usage in smaller projects and by smaller teams
+
+**Cons**: 
+- no direct way to have multiple versions of the code (achievable by using workarounds)
+- more susceptible to bugs since all features get released
+
+#### GitLab flow
+
+GitLab flow introduces environment branches to GitLab flow (e.g. staging and production).
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:68]: It's a spelling error, should be "to GitHub flow". :)

File: Git/Branching Strategies.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+11. Commit messages reflect intent.
+
+**Pros**: 
+- supports multiple versions of the code (multiple pre-production environments)
+**Cons**:
+- can be difficult to control which features go to production (e.g. code running fine in development may not work so fine in pre-production environments, so changes need to be cherry-picked)
+
+#### Trunk based development
+
+This branching strategy is explained in details [here](https://trunkbaseddevelopment.com/).
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:92]: Thanks to Marko Dosen who brought this to light in his workshop. :)

File: Git/Branching Strategies.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:68]: Yes :) Resolved.

File: Git/Branching Strategies.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+**Pros**: 
+- allows handling multiple versions of the same code (TEST, UAT, PROD...)
+
+**Cons**: 
+- complicated merging sequence
+- releases can be tricky (if branches are not merged correctly, changes need to be chosen one-by-one (cherry picked))
+- a lot of long-living branches (develop, master, release branches)
+
+#### Github flow
+
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:54]: Thought that image should be enough and self-explanatory, but I'll add something. :)

[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md]: Done, please check.

File: Git/Git Basics.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+## .gitignore
+
+.[gitignore](https://git-scm.com/docs/gitignore) file tells Git which files in your project should be ignored. It is usually (most often) located in the root directory of your project. A file content example for the .NET Core project can be found [here](https://github.com/dotnet/core/blob/master/.gitignore).
+
+Be aware that adding a certain path to .gitignore will have no effect if the file is already being tracked by Git. In this case, you can remove the file and then update the .gitignore file.
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Git Basics.md:5]: This was actually already included, I just moved it to a different file. :)

File: Git/Commits.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+Good commit should follow some basic rules:
+1. **Be meaningful and consistent**.
+	- Changes should be organized into as small groups of related changes as possible and all the groups should be committed separately (similar to Single Responsibility Principle)
+	- Author should be able to describe in simple terms what change will be introduced by applying the commit
+2. **Write a commit message using imperative present tense**.
+   	- Describes what happens if/when a particular commit is applied
+	- Past tense can also be used if that's a team's decision; in general, any of the two ways of writing a message is allowed as long as it is consistently applied and fits client and project needs
+	- NOTE 1: Present tense verbs tend to be shorter than past tense verbs, which makes messages written that way shorter
+	- NOTE 2: Imperative messages are more readable and applicable when branch/commit is reviewed before it's merged; past tense messages can be useful when reviewing past commits
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Commits.md:13]: I rewrote the section a bit. Please check and let me know what you think. :)

**Comment by andrejVuk7** [comment]
> Like I already told you, kudos for the effort! I have some styling suggestions, and I think there are a couple of grammar oddities, so I suggest running this through Grammarly.
> 
> Usually, when I write something for the Handbook, I use VS Code with the Grammarly extension (enabled for MD files, which is for some reason not the default setting). I found that to be much easier to use than c/p to Grammarly and then back :)

File: Git/Pull Requests.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+## Pull requests
+
+Definition and basic rules of a good PR can be found [here](https://infinum.com/handbook/dotnet/team-practices/new-team-members#prs).
+	
+There could also be additional pre-defined rules that every PR needs to satisfy before getting an approval (all required reviewers have approved the PR, all comments have been resolved, etc.). On top of that, all merge conflicts (concurrent changes of the same resources) should be resolved before the merge. 
+
+After merging, all changes from the source branch are available in (appended to) the target branch. Its pointer changes to point to the last commit of the pull request. Now, the source branch is obsolete and can be removed.
+
+### How to write a good PR description
```
Comment (icottiero) [inline Git/Pull Requests.md:9]: Just remembered this, should we maybe also add a section suggesting the usage of PR description templates?

**Review by icottiero — COMMENTED** [review]
Some more improvements are needed, but we're almost there! Also, besides the comments I've added, I've noticed that there are still some grammar irregularities, did you run the text through Grammarly?

File: Git/Branching Strategies.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+**Cons**: 
+
+- longer the team is working on together, the better (requires certain level of consistency in the practices of team members)
+- doesn't fully support pull requests (review of pull requests takes some time)
+
+### Summary
+
+Previously showed branching strategies are just examples of best practices that may or may not fit the needs of a specific project. Therefore, choosing the one that works the best should be done by following the already mentioned good branching strategy rules and taking into account the professional experience of team members and team processes.
+
+You can learn more about git branching [here](https://learngitbranching.js.org/) and [here](https://medium.com/@patrickporto/4-branching-workflows-for-git-30d0aaee7bf) (used as a source for this article).
```
[the remark this answers is not in the document]
Comment (andrejVuk7) [reply Git/Branching Strategies.md:143]: Added short description, please check.

File: Git/Pull Requests.md (hunk 8 above)
Comment (andrejVuk7) [reply Git/Pull Requests.md:9]: Did you mean this? 
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

**Review by icottiero — APPROVED** [review]
I'll do another round of reviewing once we deploy this to staging, but I don't expect anything significant will need fixing. Once again, good job writing all of this, this brought the Git section to a whole new level 🎉

**Review by mahrncic — APPROVED** [review]

**Pull request merged**